### ISSUE TITLE
Support for multiple exception selectors in one handler?

### ISSUE THREAD

**Issue by oubiwann** [MichaelDrogalis/dire#33]
I'd like to be able to do something like this:

```clojure
(with-handler! #'init
  [java.lang.IllegalArgumentException, java.lang.InstantiationException]
  (fn [e & args]
    (println "[ERROR] Cannot instantiate ...")
    (println (str {:args args :errors e}))))
```

A possibly naive solution jumps to mind: check i``exception-selector`` with ``vector?``, and if so, ``(map #(with-handler ... % ...) exception-selector)``

Thoughts?

(If you're okay with the proposition, I can branch & submit PR)

**Comment by MichaelDrogalis** [comment]
Seems fine, a PR would be great - thanks!

**Comment by oubiwann** [comment]
Well, I'd forgotten about the custom selectors (which are vectors), so the solution I've put together is different than the simple proposition in the ticket description. That being said, this is much less hacky (so I'm happy about that).

It did require a change to the private function ``selector-type``, though -- the check for ``:key-values`` was originally ``vector?`` and ensuring there were an even number of elements. This would obviously conflict with an even number of exceptions. So I added a third requirement: every other element must be a keyword for it to be a ``:key-values``. This then left room for a new selector type: ``:class-names``.

To pair with that, I added a new ``selector-matches?`` method that checks that the given object is an instance of at least one of the elements in the vector of exception classes.

Caveat about midje: I've only used it once before! Let me know if you'd prefer the added tests to done differently :-) (as expected, though: added tests fail without the changes and pass with them). 

Note that when I submit the PR, I'll paste some of this text there for reference ...
